account_statement_import_camt/models/parser.py

```diff
@@ -294,6 +293,14 @@ def parse_entry(self, ns, node):
             transaction["narration"],
             "%s (AddtlNtryInf)" % _("Additional Entry Information"),
         )
+        self.add_value_from_node(
+            ns,
+            node,
+            "./ns:AddtlNtryInf",
+            transaction,
+            "payment_ref",
+            join_str="\n",
+        )
```

Comment:
this is parsed into the `narration` field just above your change, so I don't think we should duplicate this data by default.

You can better add a function `amend_transaction` or such which is called at the end of this function before `generate_narration`, where the `payment_ref` field is set from some field(s) in the narration if it's unset (/). This will also be an easy hook for custom code to change this behavior to a customer's sepcs, which currently people do in `generate_narration` I guess.

Reply:
@hbrunn ,thanks for your inputs!!. I have made changes as you suggested